checks.d/kubernetes.py

```diff
@@ -235,25 +245,72 @@ def _update_container_metrics(self, instance, subcontainer, kube_labels):
+                if c_id:
+                    container_tags[c_id] = tags
+                # also store tags for aliases
+                for alias in subcontainer.get('aliases', []):
+                    container_tags[alias] = tags
+            except Exception, e:
+                self.log.error("Unable to collect metrics for container: {0} ({1}".format(c_id, e))
+
+        # container metrics from kubernetes API: limits and requests
+        for pod in pods_list['items']:
             try:
-                self._update_container_metrics(instance, subcontainer, kube_labels)
-            except Exception as e:
-                self.log.error("Unable to collect metrics for container: {0} ({1}".format(
-                    subcontainer.get('name'), e))
+                containers = pod['spec']['containers']
+                name2id = {}
+                for cs in pod['status'].get('containerStatuses', []):
+                    c_id = cs.get('containerID', '').split('//')[-1]
+                    name = cs.get('name')
+                    if name:
+                        name2id[name] = c_id
+            except KeyError:
+                self.log.debug("Pod %s does not have containers specs, skipping...", pod['metadata'].get('name'))
+                continue
+
+            for container in containers:
+                c_name = container.get('name')
+                _tags = container_tags.get(name2id.get(c_name), [])
+
+                # limits
+                try:
+                    for limit, value_str in container['resources']['limits'].iteritems():
+                        values = [float(s) for s in re.findall(r'[-+]?\d+[\.]?\d*', value_str)]
+                        if len(values) != 1:
+                            self.log.warning("Error parsing limits value string: %s", value_str)
+                            continue
+                        self.publish_gauge(self, '{}.{}.limits'.format(NAMESPACE, limit), values[0], _tags)
+                except (KeyError, AttributeError) as e:
+                    self.log.error("Unable to retrieve container limits for %s: %s", c_name, e)
+                    self.log.debug(container)
+
+                # requests
+                try:
+                    for request, value_str in container['resources']['requests'].iteritems():
+                        values = [float(s) for s in re.findall(r'[-+]?\d+[\.]?\d*', value_str)]
+                        if len(values) != 1:
+                            self.log.warning("Error parsing requests value string: %s", value_str)
+                            continue
+                        self.publish_gauge(self, '{}.{}.requests'.format(NAMESPACE, request), values[0], _tags)
+                except (KeyError, AttributeError) as e:
+                    self.log.error("Unable to retrieve container requests for %s: %s", c_name, e)
+                    self.log.debug(container)
```

Comment:
Debug logs can be very sneaky, showing up later or earlier than expected in the log file and all. I'd recommend adding a few words of description before dropping the variable like that to avoid confusion.